src/main/java/org/fundacionjala/coding/melvi/BankOcr.java

```diff
@@ -0,0 +1,266 @@
+                char9.setLength(0);
+                String number = "";
+                for (int i = 0; i < INT; i++) {
+                    if (i > 0) {
+                        line = br.readLine();
+                        if ("".equals(line)) {
+                            break;
+                        }
+                    }
+                    char1.append(line.substring(0, TRHEE));
+                    char2.append(line.substring(TRHEE, SIX));
+                    char3.append(line.substring(SIX, NINE));
+                    char4.append(line.substring(NINE, TWELVE));
+                    char5.append(line.substring(TWELVE, FIFTEEN));
+                    char6.append(line.substring(FIFTEEN, EIGHTEEN));
+                    char7.append(line.substring(EIGHTEEN, TWENTY_ONE));
+                    char8.append(line.substring(TWENTY_ONE, TWENTY_FOUR));
+                    char9.append(line.substring(TWENTY_FOUR, TWENTY_SEVEN));
+
+                }
+                number = identifyChar(char1.toString()) + identifyChar(char2.toString())
+                        + identifyChar(char3.toString()) + identifyChar(char4.toString())
+                        + identifyChar(char5.toString()) + identifyChar(char6.toString())
+                        + identifyChar(char7.toString()) + identifyChar(char8.toString())
+                        + identifyChar(char9.toString());
+
+                resultado.add(number);
+
+                line = br.readLine(); //empty separator line
+                if ("".equals(line)) {
+                    break;
+                }
+
+            }
+
+            if (br != null) {
+                br.close();
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return resultado;
+    }
+
+
+    /**
+     * @param pattern String
+     * @return result String identify the number
+     */
+    private String identifyChar(final String pattern) {
+        if (charMap.containsKey(pattern)) {
+            return charMap.get(pattern);
+        } else {
+            return "?";
+        }
+    }
+
+    /**
+     * Use story 2.
+     *
+     * @param number receive a string of numbers of 9 digits. for verify if is a account valid.
+     * @return True or False after valid the number if is a account valid.
+     */
+    public boolean validNumber(final String number) {
+        int result = 0;
+        int tam = number.length();
+        if (number.contains("?")) {
+            return false;
+        }
+        for (int i = 0; i < tam; i++) {
+            result += Integer.parseInt(number.substring(i, i + 1)) * (tam - i);
+
+        }
+        if (result % ONCE == 0) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Use story 3.
+     *
+     * @param filePath String
+     * @return ArrayList<String> processedAccounts valid an account
+     */
+    public ArrayList<String> validatedAccount(final String filePath) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)